src/Framework/QueryBuilder/Concerns/MetaQuery.php

```diff
@@ -95,7 +95,7 @@ public function attachMeta($table, $foreignKey, $primaryKey, ...$columns)
             }
 
             // Set dynamic alias
-            $tableAlias = sprintf('%s_%s_%d', ($table instanceof RawSQL) ? $table->sql : $table, 'attach_meta', $i);
+            $tableAlias = sprintf('%s_%s_%s', ($table instanceof RawSQL) ? $table->sql : $table, 'attach_meta', $columnAlias);
```

Comment:
@pauloiankoski I notice in the code that `$columnAlias` may be undefined. Make sure to do `$columnAlias ?: $column` as used elsewhere

Reply:
Resolved: 2bb06698fd5754deb5bff333dbb7e60f765195c5